File: src/C#/BehavioralPatterns/Mediator/MultipleExamples/CommunicationHub/Mediators/SensorCommandMediator.cs
using MultipleExamples.CommunicationHub.Participants;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MultipleExamples.CommunicationHub.Mediators
{
    public class SensorCommandMediator : GroupMediator
    {
        public override void Notify(Participant sender, object senderArgs)
        {
            var groupsOfParticipant = Groups.Where(gr => gr.DoesParticipantExist(sender)).ToList();

            var receivers = groupsOfParticipant
                .SelectMany(gr => gr.Participants)
                .Where(p => p != sender)
                .Distinct()
                .ToList();

            if (senderArgs == "measure")
            {
                receivers = receivers.Where(r => !(r is User)).ToList();
            }
            else
            {
                receivers = receivers.Where(r => !(r is Sensor)).ToList();
            }

            receivers.ForEach(r => r.Receive(sender, senderArgs));
        }
    }
}

File: src/PHP/BehavioralPatterns/Mediator/CommunicationHub/Mediators/SensorCommandMediator.php
<?php

namespace BehavioralPatterns\Mediator\CommunicationHub\Mediators;

use BehavioralPatterns\Mediator\CommunicationHub\Participants\Sensor;
use BehavioralPatterns\Mediator\CommunicationHub\Participants\User;

class SensorCommandMediator extends GroupMediator
{
    public function notify($sender, $senderArgs): void
    {
        $groupsOfParticipant = [];
        foreach ($this->groups as $group) {
            if ($group->doesParticipantExist($sender)) {
                $groupsOfParticipant[] = $group;
            }
        }

        $receivers = [];
        foreach ($groupsOfParticipant as $group) {
            $participants = $group->getParticipants();

            foreach ($participants as $participant) {
                if ($participant !== $sender) {
                    $receivers[] = $participant;
                }
            }
        }

        if ($senderArgs === "measure") {
            $receivers = array_filter($receivers, function ($r) {
                return !($r instanceof User);
            });
        } else {
            $receivers = array_filter($receivers, function ($r) {
                return !($r instanceof Sensor);
            });
        }

        foreach ($receivers as $receiver) {
            $receiver->receive($sender, $senderArgs);
        }
    }
}